lua/framework/math/quadratures/quadratures_lua.h

```diff
@@ -136,7 +136,7 @@ int GetProductQuadrature(lua_State* L);
  *  Example:
  * \code
  * pquad = CreateProductQuadrature(GAUSS_LEGENDRE_CHEBYSHEV,2, 1)
- * OptimizeAngularQuadratureForPolarSymmetry(pqaud, 4.0*math.pi)
+ * aquad.OptimizeForPolarSymmetry(pqaud, 4.0*math.pi)
```

Comment:
Typo: pqaud -> pquad (it was there before)

Reply:
Original code is:
```
pquad = CreateProductQuadrature(...)
OptimizeAngularQuadratureForPolarSymmetry(pqaud, 4.0*math.pi)
```
Updated code:
```
pquad = CreateProductQuadrature(...)
aquad.OptimizeForPolarSymmetry(pqaud, 4.0*math.pi)
```
i.e. using `aquad` module to work on a `pquad` handle. Notice that in lua you cannot have variable name same as module name, i.e. I cannot do:
```
aquad = CreatePorductQuadrature(...)
aquad.OptimizeForPolarSymmetry(aqaud, 4.0*math.pi)
```
if that's what you are suggesting to change here.

And I think `pquad` here refers to product quadrature. If that makes no sense I can change this to just `quad`? Unless you have a better suggestion.

Reply:
@andrsd, @zhardy-lanl  : the mistake below (of having a lua handle and then passing a dummy one because of a typo) does not result in an error. Why?
```
pquad = CreateProductQuadrature(...)
OptimizeAngularQuadratureForPolarSymmetry(pqaud, 4.0*math.pi)
```

Reply:
It should be an error. Inside `OptimizeAngularQuadratureForPolarSymmetry `, there is:
```
  auto& quadrature =
    opensn::GetStackItem<AngularQuadrature>(opensn::angular_quadrature_stack, handle, fname);
```
so if the handle is invalid, the function should throw an error (maybe via exception or somthing). If it is erroring out, then there may be some missing error checking...

Reply:
This isn't code that is getting executed, right? This is just in a function docstring. I would wager you'd get an error from `GetStackItem` if you actually ran this code.

Reply:
Unfortunately @zhardy-lanl, this is not the case. The following input runs just fine so an error check is missing somewhere. If everybody concurs, then we need to create a new issue. However, this should not stall the current PR. Ok?
```
-- Create product quadrature
pquad = CreateProductQuadrature(GAUSS_LEGENDRE_CHEBYSHEV,2,4)
OptimizeAngularQuadratureForPolarSymmetry(pqaud, 4.0*math.pi)
```

Reply:
This may be a naive question, but how would this issue get past an `at` call for a `std::vector` and a `dynamic_pointer_cast` without an issue in Release?

Reply:
I should have looked at the `GetStackItem` code. It has proper error handling. I tried the code snippet:
```
pquad = CreateProductQuadrature(GAUSS_LEGENDRE_CHEBYSHEV, 2, 4)
OptimizeAngularQuadratureForPolarSymmetry(pqaud, 4.0*math.pi)
```
And I see this error message:
```
**!**ERROR**!** LuaError: ../test/test.lua:2: attempt to call a nil value (global 'CreateProductQuadrature')
```
in Release mode.

Reply:
Ok. So the code snippet has a mistake. I need to use `aquad.CreateProductQuadrature`. With that the code runs. 

The reason why this works is this. The `aquad.CreateProductQuadrature` creates a quadrature and puts it on a stack (handle = 0). Then, we enter `aquad.OptimizeForPolarSymmetry` with `pquad` (which is `nil`), but the code does not check that and does `const int handle = lua_tointeger(L, 1);` which casts the `nil` as `0`, thus the `GetStackItem` returns the quadrature at position `0`, which is valid so all seems fine. So the code should have a check like so `if (lua_isnil(L, 1))` to handle this case...